ethcore/sync/src/chain/mod.rs

```diff
@@ -675,6 +684,55 @@ pub struct ChainSync {
     warp_sync: WarpSync,
 }
 
+#[derive(Debug, Default)]
+struct GetPooledTransactionsReport {
+    found: H256FastSet,
+    missing: H256FastSet,
+    not_sent: H256FastSet,
+}
+
+impl GetPooledTransactionsReport {
+    fn generate(
+        mut asked: Vec<H256>,
+        received: impl IntoIterator<Item = H256>,
+    ) -> Result<Self, H256> {
+        let mut out = GetPooledTransactionsReport::default();
+
+        let asked_set = asked.iter().copied().collect::<H256FastSet>();
+        let mut asked_iter = asked.drain(std::ops::RangeFull);
+        let mut txs = received.into_iter();
+        let mut next_received: Option<H256> = None;
+        loop {
+            if let Some(received) = next_received {
+                if !asked_set.contains(&received) {
```

Comment:
Seems like this could be done with a set difference (`received - asked != 0`) more efficiently, afaict from the context where `generate` is being used, we do have a set of `imported` transactions already.